### DISCUSSION TITLE
How to select points from the 3d display of gl.GLScatterPlotItem

### DISCUSSION THREAD

**Discussion by wanghui-min** [pyqtgraph/pyqtgraph#2569] [Q&A]
How to select points from the 3d display of gl.GLScatterPlotItem

**Comment by jcfuentes** [comment]
if anyone else needs the solution, looks like this (posted originally by Wanghui-min):


    def __choose_detection(
        self,
        ogl_item: gl.items.GLScatterPlotItem.GLScatterPlotItem,
        region,
        gl_vw: GLViewWidget,
    ):
        view_w = gl_vw.width()
        view_h = gl_vw.height()
        points = ogl_item.pos

        m = gl_vw.projectionMatrix() * gl_vw.viewMatrix()
        m = np.array(m.data(), dtype=np.float32).reshape((4, 4))
        one_mat = np.ones((points.shape[0], 1))
        points = np.concatenate((points, one_mat), axis=1)
        new = np.matmul(points, m)
        new[:, :3] = new[:, :3] / new[:, 3].reshape(-1, 1)
        new = new[:, :3]
        projected_array = np.zeros((new.shape[0], 2))
        projected_array[:, 0] = (new[:, 0] + 1) / 2
        projected_array[:, 1] = (-new[:, 1] + 1) / 2
        projected_array[:, 0] = projected_array[:, 0] - (region[0] / view_w)
        projected_array[:, 1] = projected_array[:, 1] - (region[1] / view_h)

        distance_array = np.power(
            np.power(projected_array[:, 0], 2) + np.power(projected_array[:, 1], 2), 0.5
        )
        min_index = np.nanargmin(distance_array)

        logging.debug(
            "focusID: Det %d- %s, %s", min_index, str(ogl_item.pos[min_index]), "x"
        )

**Reply by dx-momo** [reply]
Hello, may I ask which version of pyqtgraph you are using

**Reply by jcfuentes** [reply]
0.13.3 and PyQt5, it does not work well with PySide6, I wanted to try PyQt6 today.

**Reply by dx-momo** [reply]
好吧，我已经做出来了，你需要代码吗
